### PR TITLE
Let the Triple struct determine the host triple automatically

### PR DESCRIPTION
**Pull request by jakepetroules**
...instead of hardcoding a bunch of triples and selecting what is in
most cases only a guess/approximation.

This also more closely models the fields of a target triple based on how
LLVM looks at it.

### CODE REVIEW

**Comment by jakepetroules** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@jrose-apple, @ddunbar: can you review this?

File: Sources/Build/Triple.swift [hunk 1]
```diff
@@ -22,61 +22,155 @@ public struct Triple {
...
-        case apple
+
+        /// Indicates Apple Inc., the vendor of iOS, macOS, tvOS, and watchOS.
+        case Apple = "apple"
+
+        /// Indicates vendor-agnostic platform "PC"-type platform.
+        case PC = "pc"
+
+        /// Indicates Sony Computer Entertainment, Inc., the vendor of PS4 OS.
+        case SCEI = "scei"
```
Comment (ddunbar) [inline Sources/Build/Triple.swift]: I would keep the previous casing, I prefer downcasing initial acronyms (although I think Swift hasn't formally decided on a convention).

Comment (jakepetroules) [reply Sources/Build/Triple.swift]: > NOTE: There are some subtle differences between Swift's expected -target triples and LLVM's triples.

For example?

Comment (ddunbar) [reply Sources/Build/Triple.swift]:
```
$ swiftc -c -### t.swift
.... -target x86_64-apple-darwin17.6.0 ...

$ clang -c -### t.c
... "-triple" "x86_64-apple-macosx10.13.0" ...
```

Comment (jrose-apple) [reply Sources/Build/Triple.swift]: They're interchangeable enough. Swift canonicalizes pretty aggressively internally when it cares.

Comment (jrose-apple) [reply Sources/Build/Triple.swift]: Note that "darwin" in an LLVM triple, whether Clang or Swift, is assumed to mean "macosx", not "any Apple platform".

File: Sources/Build/Triple.swift [hunk 2]
```diff
@@ -87,55 +181,193 @@ public struct Triple {
...
+extension Triple {
+    /// Creates a triple with the given architecture, OS and OS version.
+    /// The vendor and ABI fields are determined automatically.
+    public static func create(arch: Arch, os: OS, osVersion: Version? = nil) -> Triple {
+        let vendor: Vendor = {
+            switch os {
+            case .Darwin, .iOS, .macOS, .tvOS, .watchOS:
+                return .Apple
+            case .FreeBSD, .Haiku, .Linux, .Windows:
+                return .PC
```
Comment (jrose-apple) [inline Sources/Build/Triple.swift]: This is a behavior change from before, where we used "unknown". Are you sure this is correct?

Comment (compnerd) [reply Sources/Build/Triple.swift]: This is fun actually.  It depends on the Linux distribution.  `pc`  is the legacy vendor that is used by default on most i686 targets.  `unknown` is the proper vendor technically.

**Comment by jakepetroules** [comment]
@swift-ci please smoke test

**Comment by jakepetroules** [comment]
@swift-ci please smoke test

**Comment by jakepetroules** [comment]
@swift-ci please smoke test

**Comment by jakepetroules** [comment]
@swift-ci please smoke test

**Comment by futurejones** [comment]
Currently there are several failing swiftpm tests on Linux/AArch64 systems. These have a host triple of "aarch64-unknown-linux". The tests are failing due to the hard coding of "-target", "x86_64-unknown-linux" as the default.
This PR fixes this issue and all tests for swiftpm on Linux/AArch64 are now passing.

**Comment by tomerd** [comment]
@jakepetroules I think this can be now closed? 

cc @neonichu

**Comment by jakepetroules** [comment]
There's still some useful stuff in here, but it's so far out of date I'd need to start over. Closing for now.

**Pull request closed without merging**